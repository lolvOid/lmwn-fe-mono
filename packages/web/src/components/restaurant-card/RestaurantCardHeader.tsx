type RestaurantCardHeaderProps = {
    name: string;
    openTime: string;
    closeTime: string;
};

const RestaurantCardHeader = ({ name, openTime, closeTime }: RestaurantCardHeaderProps) => {
    return (
        <div className="bg-gray-900 rounded-t-2xl text-white lg:max-w-[600px] -my-3 lg:-my-4 mx-auto w-full px-8 lg:py-4 py-2 sticky z-[400] flex flex-col">
            <span className="text-lg">{name || 'Restaurant Name'}</span>
            <span className="text-sm text-yellow-300 line">
                Open {openTime || '00:00'} - Close {closeTime || '00:00'}
            </span>
        </div>
    );
};

export default RestaurantCardHeader;
