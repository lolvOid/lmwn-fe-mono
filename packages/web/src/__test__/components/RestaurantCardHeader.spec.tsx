import { render, screen } from '@testing-library/react';
import RestaurantCardHeader from '@/components/restaurant-card/RestaurantCardHeader'; // Assuming your component is in './RestaurantCardHeader'
import renderer from 'react-test-renderer';

describe('RestaurantCardHeader', () => {
    it('Should render restaurant name and open/close hours correctly', () => {
        const name = 'My Restaurant';
        const openTime = '10:00';
        const closeTime = '20:00';

        render(<RestaurantCardHeader name={name} openTime={openTime} closeTime={closeTime} />);

        expect(screen.getByText(name)).toBeInTheDocument();
        expect(screen.getByText(`Open ${openTime} - Close ${closeTime}`)).toBeInTheDocument();
    });

    it('Should render with default values when props are not provided', () => {
        render(<RestaurantCardHeader name={''} openTime={''} closeTime={''} />);

        expect(screen.getByText('Restaurant Name')).toBeInTheDocument();
        expect(screen.getByText('Open 00:00 - Close 00:00')).toBeInTheDocument();
    });

    it('Should render and match the default snapshot', () => {
        const header = renderer
            .create(
                <RestaurantCardHeader
                    name={'My Restaurant'}
                    openTime={'7:00'}
                    closeTime={'20:00'}
                />,
            )
            .toJSON();

        expect(header).toMatchSnapshot();
    });
});
