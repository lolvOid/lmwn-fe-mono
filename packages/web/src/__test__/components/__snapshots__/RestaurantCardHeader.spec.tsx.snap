// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCardHeader Should render and match the default snapshot 1`] = `
<div
  className="bg-gray-900 rounded-t-2xl text-white lg:max-w-[600px] -my-3 lg:-my-4 mx-auto w-full px-8 lg:py-4 py-2 sticky z-[400] flex flex-col"
>
  <span
    className="text-lg"
  >
    My Restaurant
  </span>
  <span
    className="text-sm text-yellow-300 line"
  >
    Open 
    7:00
     - Close 
    20:00
  </span>
</div>
`;
